src/repository/cert.rs

```diff
@@ -2646,6 +2646,33 @@ mod test {
         ).unwrap();
     }
 
+    #[test]
+    fn signature_algorithm_mismatch() {
+        let roa = crate::repository::roa::Roa::decode(
+            include_bytes!(
+                "../../test-data/example-ripe.roa"
+            ).as_ref(),
+            false
+        ).unwrap();
+        assert!(
+            roa.cert().inspect_ee_at(
+                true, Time::utc(2020, 05, 01, 0, 0, 0)
+            ).is_ok()
+        );
+
+        let mft = crate::repository::manifest::Manifest::decode(
+            include_bytes!(
+                "../../test-data/signature-alg-mismatch.mft"
+            ).as_ref(),
+            false
+        ).unwrap();
+        assert!(
+            mft.cert().inspect_ee_at(
+                true, Time::utc(2020, 05, 01, 0, 0, 0)
+            ).is_err()
+        );
```

Comment:
Is this testing the new with/without parameter support added in this PR, or is this testing for a possible malformed error (as returned on line 123)? If possible/trivial, being more specific than just `.is_err()` could sort of self-document this test a bit better, and make it more robust as well. (Even though the mft filename hints at what is going on here)

Reply:
Given that the error is really a unit struct, `is_err` should be sufficient here. I will add a comment, though.

Reply:
That helps a lot, thanks!